### PR TITLE
test(lib/runtime): Add new test data and use it in a test to get async backing params

### LINKED ISSUE #3809: test(lib/runtime): `TestInstance_ParachainHostAsyncBackingParams` logs an error
## Issue summary

- In `feat/parachain` branch, When we run the `TestInstance_ParachainHostAsyncBackingParams` test, it passes, but it also logs an error saying the corrupted state
```
axay@axay-ThinkPad-E14-Gen-3:~/github/chainsafe/gossamer$ go test ./lib/runtime/wazero/ -v -run TestInstance_ParachainHostAsyncBackingParams
=== RUN   TestInstance_ParachainHostAsyncBackingParams
=== PAUSE TestInstance_ParachainHostAsyncBackingParams
=== CONT  TestInstance_ParachainHostAsyncBackingParams
2024-03-25T20:10:06+05:30 CRITICAL target=runtime::storage message=Corrupted state at `0x06de3d8a54d27e44a9d5ce189618f22db4b49d95320d9021994c850f25b8e385`: Error	ext_logging_log_version_1	pkg=runtime module=wazero

--- PASS: TestInstance_ParachainHostAsyncBackingParams (1.12s)
PASS
ok  	github.com/ChainSafe/gossamer/lib/runtime/wazero	1.177s
```
## Other information and links

- merge the solution in `feat/parachain` branch

### PR DESCRIPTION
**Pull request by axaysagathiya**
## Changes
added new test data

## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Issues
Closes #3809 

## Primary Reviewer

@kishansagathiya @EclesioMeloJunior

### CODE REVIEW

File: lib/runtime/wazero/instance_test.go [hunk 1]
```diff
@@ -38,6 +38,9 @@ import (
 //go:embed testdata/parachain.yaml
 var parachainTestDataRaw string
 
+//go:embed testdata/parachain2.yaml
+var parachain2TestDataRaw string
+
```
Comment (EclesioMeloJunior) [inline lib/runtime/wazero/instance_test.go]: could you change to parachains_configuration_v180.yaml? then we know what is the purpose of the yaml file

File: lib/runtime/wazero/instance_test.go [hunk 2]
```diff
@@ -1617,3 +1633,16 @@ func getParachainHostTrie(t *testing.T) *trie.Trie {
 
 	return tt
 }
+
+func getParachainHostTrie2(t *testing.T) *trie.Trie {
```
Comment (EclesioMeloJunior) [inline lib/runtime/wazero/instance_test.go]: I would suggest to change the function `getParachainHostTrie` to receive another param called `storage []Storage`

```go
func getParachainHostTrie(t *testing.T, testDataStorage []Storage) *trie.Trie {
	tt := trie.NewEmptyTrie()
	for _, s := range testDataStorage {
		key := common.MustHexToBytes(s.Key)
		value := common.MustHexToBytes(s.Value)
		err := tt.Put(key, value)
		require.NoError(t, err)
	}

	return tt
}

getParachainHostTrie(t, parachainTestData.Storage)
getParachainHostTrie(t, parachain2TestData.Storage)
```

**Review by EclesioMeloJunior — DISMISSED** [review]

File: lib/runtime/wazero/instance_test.go [hunk 3]
```diff
@@ -38,6 +37,9 @@ import (
 //go:embed testdata/parachain.yaml
 var parachainTestDataRaw string
 
+//go:embed testdata/parachains_configuration_v180.yaml
+var parachainTestData2Raw string
```
Comment (EclesioMeloJunior) [inline lib/runtime/wazero/instance_test.go]: Also change the name to match the file name

**Review by EclesioMeloJunior — APPROVED** [review]
LGTM, just change the variable name for the embedded file

**Review by kishansagathiya — APPROVED** [review]

**Pull request merged**